Visual Studio and Visual Studio for Mac overwrite each other's changes in the solution

I've seen different project type GUIDs used for C# projects in VS 2017 solutions. 

dotnet/sdk and dotnet/cli template and use `FAE04EC0-301F-11D3-BF4B-00C04F79EFBC`. cref https://github.com/Microsoft/msbuild/pull/1607, https://github.com/dotnet/sdk/issues/522

Yet, VS 2017 will sometimes automatically change solution files using `FAE04EC0..` to a different GUID: `9A19103F-16F7-4668-BE54-9A1E7A4F7556`. cref https://github.com/aspnet/Logging/pull/577#discussion_r107281676

I took a peek [at the code](https://github.com/dotnet/roslyn-project-system/blob/f71940e85c15f127a1e981db158dbb8296c57003/src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/Packaging/CSharpProjectSystemPackage.cs#L37-L38) and found this:
```c#
        public const string ProjectTypeGuid = "9A19103F-16F7-4668-BE54-9A1E7A4F7556";
        public const string LegacyProjectTypeGuid = "FAE04EC0-301F-11d3-BF4B-00C04F79EFBC";
```

So, which one is "right"?

cc @davkean @dsplaisted

I have a big problem here.  I was just upgrading an existing 4.6.1 .NET framework with ASPNET Core to 4.7.1.  I started getting errors about mixed frameworks in the packages.config.  A solution mentioned was to right mouse click on the project and covert it to package references.  This fixed the build errors (VS 15.7.6).  But now I am getting weird build errors on my TFS build server like this: "error CS0234: The type or namespace name 'Extensions' does not exist in the namespace 'Microsoft'". I am running with build agent 2.136.  I have narrowed it down to any project that I did this convert to PackageReference to actually had already a '9A19103F-16F7-4668-BE54-9A1E7A4F7556' project type (we were tying to do a .NET Core app, but had to fall back to .Net Framework because of 3rd party stuff) and the build server is not dropping DLLs from the Nuget packages into those project folders causing these weird build errors.

So I am now going to try and change all project files to 'FAE04EC0-301F-11d3-BF4B-00C04F79EFBC' and see if I can (most likely manually for several project files) get this project to build.  Hope this helps someone.

--- a/eng/imports/Versions.props
+++ b/eng/imports/Versions.props
@@ -1,6 +1,6 @@
 <!-- Any changes to this file or format requires updates in project-system-vscode -->
 <Project>
   <PropertyGroup>
-    <CPSPackageVersion>17.11.207</CPSPackageVersion>
+    <CPSPackageVersion>17.12.21-pre</CPSPackageVersion>
   </PropertyGroup>
 </Project>

--- a/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Packaging/ProjectTypeRegistration.cs
+++ b/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Packaging/ProjectTypeRegistration.cs
@@ -15,6 +15,7 @@ using Microsoft.VisualStudio.ProjectSystem.VS;
     Capabilities = ProjectTypeCapabilities.VisualBasic,
     DisableAsynchronousProjectTreeLoad = true,
     PossibleProjectExtensions = "vbproj",
+    PreferredPersistProjectTypeGuid = ProjectType.LegacyVisualBasic,
     NewProjectRequireNewFolderVsTemplate = true,
     SupportsCodespaces = true,
     SupportsSolutionChangeWithoutReload = true)]
@@ -32,6 +33,7 @@ using Microsoft.VisualStudio.ProjectSystem.VS;
     Capabilities = ProjectTypeCapabilities.FSharp,
     DisableAsynchronousProjectTreeLoad = true,
     PossibleProjectExtensions = "fsproj",
+    PreferredPersistProjectTypeGuid = ProjectType.LegacyFSharp,
     NewProjectRequireNewFolderVsTemplate = true,
     SupportsCodespaces = true,
     SupportsSolutionChangeWithoutReload = true)]
@@ -49,6 +51,7 @@ using Microsoft.VisualStudio.ProjectSystem.VS;
     Capabilities = ProjectTypeCapabilities.CSharp,
     DisableAsynchronousProjectTreeLoad = true,
     PossibleProjectExtensions = "csproj",
+    PreferredPersistProjectTypeGuid = ProjectType.LegacyCSharp,
     NewProjectRequireNewFolderVsTemplate = true,
     SupportsCodespaces = true,
     SupportsSolutionChangeWithoutReload = true)]

